rust-symcrypt/src/gcm.rs

```diff
@@ -149,15 +151,22 @@ impl GcmExpandedKey {
             convert_cipher(cipher),
         )?;
         let gcm_expanded_key = GcmExpandedKey {
-            expanded_key: expanded_key,
+            expanded_key,
             key_length: key.len(),
         };
         Ok(gcm_expanded_key)
     }
 
-    /// `encrypt_in_place` takes in a `&mut buffer` that has the plain text data to be encrypted. After the encryption has been completed,
-    /// the `buffer` will be over-written to contain the cipher text data. `encrypt_in_place` will also take in `tag` which is
-    /// a `&mut buffer` where the resulting tag will be written to.
+    /// `encrypt_in_place` performs an in-place encryption on the `&mut buffer` that is passed. This call cannot fail.
+    ///
+    /// `nonce` is a `&[u8; 12]` that is used as the nonce for the encryption.
+    ///
+    /// `auth_data` is an optional `&[u8]` that can be provided, if you do not wish to provide any auth data, input an empty array.
+    ///
+    /// `buffer` is a `&mut [u8]` that contains the plain text data to be encrypted. After the encryption has been completed,
+    /// `buffer` will be over-written to contain the cipher text data.
+    ///
+    /// `tag` is a `&mut [u8]` which is the buffer where the resulting tag will be written to.
```

Comment:
>/// `tag` is a `&mut [u8]` which is the buffer where the resulting tag will be written to. [](http://example.com/codeflow?start=4&length=90)

Provide information on tag length. From symcrypt:
//      - cbTag: size of tag. cbTag must be one of {12, 13, 14, 15, 16} per SP800-38D
//          section 5.2.1.2. The optional shorter tag sizes (4 and 8) are not supported.

Reply:
Added